src/arch/riscv/isa/templates/vector_arith.isa

```diff
@@ -129,12 +129,55 @@ Fault
     if (machInst.vill)
         return std::make_shared<IllegalInstFault>("VILL is set", machInst);
 
-    %(op_decl)s;
-    %(op_rd)s;
-    %(vm_decl_rd)s;
-    %(copy_old_vd)s;
-    %(code)s;
-    %(op_wb)s;
+    auto SEW = vtype_SEW(vtype);
+
+    switch (SEW) {
+      case 8: {
+        using vext  [[maybe_unused]] = int8_t;
+        using vextu [[maybe_unused]] = uint8_t;
+        %(op_decl)s;
+        %(op_rd)s;
+        %(vm_decl_rd)s;
+        %(copy_old_vd)s;
+        %(code)s;
+        %(op_wb)s;
+        break;
+      }
+      case 16: {
+        using vext  [[maybe_unused]] = int16_t;
+        using vextu [[maybe_unused]] = uint16_t;
+        %(op_decl)s;
+        %(op_rd)s;
+        %(vm_decl_rd)s;
+        %(copy_old_vd)s;
+        %(code)s;
+        %(op_wb)s;
+        break;
+      }
+      case 32: {
+        using vext  [[maybe_unused]] = int32_t;
+        using vextu [[maybe_unused]] = uint32_t;
+        %(op_decl)s;
+        %(op_rd)s;
+        %(vm_decl_rd)s;
+        %(copy_old_vd)s;
+        %(code)s;
+        %(op_wb)s;
+        break;
+      }
+      case 64: {
+        using vext  [[maybe_unused]] = int64_t;
+        using vextu [[maybe_unused]] = uint64_t;
+        %(op_decl)s;
+        %(op_rd)s;
+        %(vm_decl_rd)s;
+        %(copy_old_vd)s;
+        %(code)s;
+        %(op_wb)s;
+        break;
+      }
+      default: break;
+    }
```

Comment:
The `VectorIntMicro*` templates will produce a c++ template class. The value of sew can be passed as the parameter of template class. e.g. `Vadd_vMicro<uint8_t>`
Since the `vextu` and `vext` postfixes are only used in vssra and vssrl instructions, maybe you can add special isa formats and templates for them to keep `VectorIntMicro*` templates tidy.

Reply:
Fixed. It didn't actually need the special handling. Using `vi` instead of `vext` was sufficient:

```
                    0x2b: vssra_vi({{
                        int sh = SIMM5 & (sew - 1);
                        __int128_t val = Vs2_vi[i];```

Reply:
Yeah, it looks better.